DEVELOPMENT.md

```diff
@@ -379,6 +379,177 @@ Comments can also be used to indicate steps to take in the future (*TODOs*). Suc
+
+##### View component checklist
+
+- Conciseness and no stateful logic (use containers instead)
+- Uses `<PageTitle />` component to define heading and page title
+- Uses breadcrumbs (if within breadcrumb view)
+- Fetching necessary data (via `withRequest` HOC), if not done by a container
+- Unavailable "catch-all"-routes are caught by `<NotFoundRoute />` component, including subviews
+- Errors should be caught by the `<ErrorView />` error boundary component
+- `withFeatureRequirement` HOC is used to prevent access to routes that the user has no rights for
+- Ensured responsiveness and usage of the grid system
+
+#### Utility components
+
+These components do not render any DOM elements and are hence not *visible* by themselves. Utility components can be higher order components or similar components, that modify their children or introduce a side effect to the render tree.
+
+To decide whether a component is a utility component ask yourself:
+- Is this component a higher order component?
+- Is this component invisible on its own?
+- Is this component an abstraction layer on top of another component?
+
+If you can answer at least one of those questions with yes, then you likely have a container component.
+
+#### Global or Application Scope?
+
+Components can be categorized as either local (e.g. `pkg/webui/{console|oauth}/{components|containers}`) or global (e.g. `pkg/webui/{components|containers}`). The distinction should come naturally: Global components are ones that can be used universally in every application. Local components are tied to a specific use case inside the respective application.
+
+Sometimes, you might find that during implementing an application specific component that it can actually be generalized without much refactoring and hence be a useful addition to our global component library.
+
+### Frontend Related Pull Requests
+
+Pull requests for frontend related changes generally follow our overall pull request scheme. However, in order to assist reviewers, a browser screenshot of the changes is included in the PR comment, if applicable.
+
+It might help you to employ the following checklist before opening the pull request:
+
+* [ ] All messages [translated](#frontend-translations)?
+* [ ] No large files (e.g. unminified SVG assets)?
+* [ ] Screenshot in PR description?
+* [ ] Responsiveness checked?
+* [ ] New components [categorized correctly](#global-or-application-scope) (global/local, container/component)?
+* [ ] Feature flags added (if applicable)?
+* [ ] All views use `<IntlHelmet />` properly?
+* [ ] Storybook story added / updated?
+* [ ] Prop types / default props added?
```

Comment:
I'd fix this in this PR, but it would mean a lot of refactoring in the OAuth app, which will be replaced soon. Until then, I'd suggest to keep this item in there.